src/app/checkout/Checkout.tsx

```diff
@@ -138,7 +138,15 @@ class Checkout extends Component<CheckoutProps & WithCheckoutProps & WithLanguag
             const errorFlashMessages = data.getFlashMessages('error') || [];
 
             if (errorFlashMessages.length) {
-                this.setState({ error: new Error(errorFlashMessages[0].message) });
+                const message = errorFlashMessages[0].message;
+                const title = errorFlashMessages[0].title ? errorFlashMessages[0].title : null;
+
+                if (title) {
+                    const errorData = {data: {}, message, title, name: 'default'};
```

Comment:
you could simplify this by always setting a custom error?
```js
this.setState({ error: new CustomError({
  title: errorFlashMessages[0].title,
  message: errorFlashMessages[0].message,
  data: {},
  name: 'default'
})});
```

Reply:
We are using the interface 
```js
export interface CheckoutState {
    activeStepType?: CheckoutStepType;
    customerViewType?: CustomerViewType;
    defaultStepType?: CheckoutStepType;
    error?: Error | CustomError;
    flashMessages?: FlashMessage[];
    isMultiShippingMode: boolean;
    isCartEmpty: boolean;
    isRedirecting: boolean;
    hasSelectedShippingOptions: boolean;
}

```

then I left the your approach we got conflict with the property **title** and it's expected the type **Error** in others parts of the code.

Reply:
I tried to implement the change and I couldn't see any error? `CustomError` is a subclass of Error so that should be fine?

<img width="1360" alt="Screen Shot 2020-10-28 at 12 24 22 pm" src="https://user-images.githubusercontent.com/1621894/97379278-908cfb00-1918-11eb-9063-902722a3a1f4.png">
<img width="813" alt="Screen Shot 2020-10-28 at 12 24 36 pm" src="https://user-images.githubusercontent.com/1621894/97379280-92ef5500-1918-11eb-8ff2-25d662a66831.png">